### PR TITLE
v0.26.1 and implement local dumping

### PR DESCRIPTION
**Pull request by silasdavis**
This PR brings local dumping - as in just dump from a burrow directory rather than starting a chain and dumping over GRPC. It makes things a bit less fiddly and avoids extra pointless CPU usage on a sometimes CPU-bound task (dumping).

The refactor also enable dumping from a particular height which opens the way for resumption #1142 

There is also some misc house-keeping in here:
- Avoiding panics by always checking for tx exceptions in burrow deploy
- Fix vent options to make --txs and --blocks have an effect

### CODE REVIEW

File: cmd/burrow/commands/config_options.go [hunk 1]
```diff
@@ -20,14 +20,30 @@ type configOptions struct {
...
+
+var configFileOption = cli.StringOpt{
+	Name:   "c config",
+	Desc:   "Use the specified burrow config file",
+	EnvVar: "BURROW_CONFIG_FILE",
+}
+
+const genesisFileSpec = "[--genesis=<genesis json file>]"
+
+var genesisFileOption = cli.StringOpt{
```
Comment (gregdhill) [inline cmd/burrow/commands/config_options.go]: @silasdavis why move these here?

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,8 @@ SOLIDITY_GO_FILES = $(patsubst %.sol, %.sol.go, $(SOLIDITY_FILES))
 
 CI_IMAGE="hyperledger/burrow:ci"
 
+BIN_PATH?=${HOME}/.local/bin
+
```
Comment (seanyoung) [inline Makefile]: Why install stuff into ~/.local/bin and what was wrong with the old scheme?

File: binary/integer.go [hunk 3]
```diff
@@ -28,38 +28,34 @@ var tt255 = new(big.Int).Lsh(big1, 255)
...
-func PutUint64LE(dest []byte, i uint64) {
-	binary.LittleEndian.PutUint64(dest, i)
-}
-
-func GetUint64LE(src []byte) uint64 {
-	return binary.LittleEndian.Uint64(src)
-}
-
-func PutUint64BE(dest []byte, i uint64) {
+func PutUint64(dest []byte, i uint64) {
```
Comment (seanyoung) [inline binary/integer.go]: If I see a uint64 to bytes conversion, I would like to know what endianness it is. Then again most of these functions are a thin wrapper around encoding/binary so maybe they should go away instead.

File: cmd/burrow/commands/config_options.go (hunk 1 above)
Comment (silasdavis) [reply cmd/burrow/commands/config_options.go]: Just so I can reuse them in dump

File: Makefile (hunk 2 above)
Comment (silasdavis) [reply Makefile]: Now we are on modules and with Go removing the necessity to even set up `GOPATH` it seems better to make people set up or care about `GOPATH` anymore

File: binary/integer.go (hunk 3 above)
Comment (silasdavis) [reply binary/integer.go]: Yeah possibly...

My thinking here was that insofar as the binary package does anything it establishes a convention that if we just want some bytes for for an integer then it is big endian. If you want/need to think about endianess then you should be using binary directly.

They do provide a little bit of convenience around directly returning the bytes, casting etc. I _think_ I'd rather have them here than strewn around the rest of code, but open to persuasion.

File: Makefile (hunk 2 above)
Comment (silasdavis) [reply Makefile]: Open to other suggestions for installing binaries, but `.local/bin` is part of this spec: https://www.freedesktop.org/software/systemd/man/file-hierarchy.html - and happens to be what I use for rootless installs.

Comment (seanyoung) [reply Makefile]: If GOPATH is unset, it defaults to $HOME/go

https://golang.org/cmd/go/#hdr-GOPATH_environment_variable

Comment (seanyoung) [reply Makefile]: go install will go to $GOPATH/bin (which defaults to $HOME/go) and make install will go $HOME/.local/bin 
That's confusing. Best not do anything surprising for users.

File: binary/integer.go (hunk 3 above)
Comment (seanyoung) [reply binary/integer.go]: "establishes the convention" just means it's confusing for anyone new to our code and does not improve readability.

File: Makefile (hunk 2 above)
Comment (silasdavis) [reply Makefile]: Okay, I will default to `$GOPATH/bin` and set default fo GOPATH

File: binary/integer.go (hunk 3 above)
Comment (silasdavis) [reply binary/integer.go]: let's remove the binary helpers next time we touch this

**Comment by silasdavis** [comment]
CI is playing up in strange ways - we'll merge this regardless if I can get a second confirmation that `make test_integration` passes locally

**Pull request merged**